app/src/main/java/com/quran/labs/androidquran/ui/PagerActivity.java

```diff
@@ -1837,6 +1887,217 @@ public void onError(@NonNull Throwable e) {
     );
   }
 
+  public void shareAyahAudio(SuraAyah start, SuraAyah end) {
+    audioCacheFilePaths.clear();
+
+    kotlin.Pair pair2 = getReorderedAyatPair(start, end);
+    selectedStartSuraAyah = (SuraAyah) pair2.component1();
+    selectedEndSuraAyah = (SuraAyah) pair2.component2();
+
+    selectedQari = audioStatusBar.getAudioInfo();
+    AudioPathInfo audioPathInfo = audioUtils.getLocalAudioPathInfo(selectedQari);
+
+    assert audioPathInfo != null;
+    boolean gaplessDatabaseExists = audioPathInfo.getGaplessDatabase() != null;
+
+    if (gaplessDatabaseExists) {
+      if (audioFilesExist(audioPathInfo)) {
+        createAndShareAudio(selectedStartSuraAyah, selectedEndSuraAyah, audioPathInfo);
+      } else {
+        requestDownload(audioPathInfo);
+      }
+    }
+  }
+
+  private kotlin.Pair getReorderedAyatPair(SuraAyah start, SuraAyah end) {
+    kotlin.Pair pair;
+    if (start.compareTo(end) <= 0) {
+      pair = TuplesKt.to(start, end);
+    } else {
+      Timber.Forest.e(
+          new IllegalStateException("End isn't larger than the start: " + start + " to " + end));
+      pair = TuplesKt.to(end, start);
+    }
+    return pair;
+  }
```

Comment:
```suggestion
  private kotlin.Pair<SuraAyah, SuraAyah> getReorderedAyatPair(SuraAyah start, SuraAyah end) {
    return if (start >= end) {
        start to end
    } else {
       end to start
    }
  }
```

when we move a lot of this logic, we can also change kotlin.Pair to just pair

Reply:
The suggestion would be good if `PagerActivity` was in kotlin. I tried to simply it but it could not get simpler in java